### ISSUE TITLE
db passwords with characters (specifically @ and possibly others) will break pub/sub due to the connection string 

### ISSUE THREAD

**Issue by wezell** [dotCMS/core#21363]
Versions of dotCMS that use pub/sub, some special characters (specifically @ and possibly others) will break pub/sub due to the connection string it uses without escaping those characters. 

We need to update the way dotCMS generates the pub/sub connection string and escape special characters.

Example that will break current pub/sub:

Create a db user
```
psql > create user testing with superuser password 'sdfFtBEg5435j!XRb%SLxG@?'
```

Then update your connection props
```
driverClassName=org.postgresql.Driver
jdbcUrl=jdbc:postgresql://dbserver/dotcms
username=testing
password=sdfFtBEg5435j!XRb%SLxG@?
```

**Comment by wezell** [comment]
this should be backported to 21.06 LTS

**Comment by bryanboza** [comment]
Fixed, tested on release-22.02 // Docker. // FF

**Comment by clemente-biondo** [comment]
+1 for the backport to 21.06 LTS
This bug is hitting us hard on production. 
We have a dotcms cluster (2 nodes) on Azure paired with an Azure database for Postgres.
The problem is that every Azure database for Postgres username contains a "@" so we cant simply avoid the problem by creating a different username.
